js/parser.js

```diff
@@ -153,6 +159,14 @@
         this._catalog = catalogParts[0];
         this._version = catalogParts[1] || null;
 
+        if (catalogObject) {
```

Comment:
Isn't it the case that the catalog is always used now? Didn't you make changes to each calling site?

Reply:
Internally in ermrestjs, this function is used without passing the `catalog` object. For example, the `resolve` function is using this to figure out which catalog it should use, so it doesn't have access to the `catalogObject` yet and cannot pass it.